src/Mvc/Mvc.Formatters.Xml/src/XmlSerializerInputFormatter.cs

```diff
@@ -99,39 +98,43 @@ public virtual InputFormatterExceptionPolicy ExceptionPolicy
             }
 
             var request = context.HttpContext.Request;
-
+            Stream readStream = new NonDisposableStream(request.Body);
```

Comment:
Why not `var`?

Reply:
We assign this to a FileBufferingReadStream in the if, so cannot be an instance of NonDisposableStream